src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComObject.cs

```diff
@@ -23,8 +23,7 @@ public sealed unsafe class ComObject : IDynamicInterfaceCastable, IUnmanagedVirt
 
         private readonly object? _runtimeCallableWrapper;
 
-        // This is an int so we can use the Interlocked APIs to update it.
-        private volatile int _released;
+        private volatile bool _released;
```

Comment:
I assume bool overloads being slower on some platforms doesn't matter here?

Reply:
Which platforms?

Reply:
RISC-V will be significantly impacted since their Atomics ISA only provides 4B and 8B atomics so 1B and 2B have to be emulated (they've recently ratified a separate ISA for 1B and 2B atomics but it'll take a while for hardware to get it). Other platforms all have small atomics I think but only X86/X64 and ARM64 (and ARM32 in #97792) have it implemented today and even on those platforms they're slightly more expensive due to requiring zero-extensions.

Reply:
Thanks. That seems fine. As noted in the PR where the new Interlocked APIs were added, if we're happy to expose APIs for the smaller data types, we should be happy to use them everywhere they're applicable. Otherwise we're just creating a pit of failure by exposing APIs that are more attractive / relevant but then saying they shouldn't be used.

Reply:
Perhaps, by the time RISC-V gets more popular it will get an extension to do byte atomics natively, for now, it feels a bit odd to trade readability over peformance (overhead is not clear, very much could be invisible) on a single non-mainstream platforms to be fair. 

Example: https://godbolt.org/z/dovM1va3v

Reply:
> it feels a bit odd to trade readability over peformance

Meaning you agree using bool instead of int is desirable, yes?

Reply:
Well, we either never use `Interlocked.X` for `bool` for the same reason (why did we expose it then?) or use everywhere where appropriate, but it's just my opinion, I don't know how bad the impact potentially is 🙂